java/ql/src/utils/stub-generator/Stubs.qll

```diff
@@ -404,20 +400,45 @@ private string stubMember(Member m) {
 
 private string stubAnnotation(Annotation a) {
   if exists(a.getAValue())
-  then result = a.toString() + "(" + concat(stubAnnotationValue(a.getAValue()), ",") + ")"
-  else result = a.toString()
+  then
+    result =
+      a.getType().getName() + "(" +
+        concat(string name, Expr value |
+          value = a.getValue(name)
+        |
+          name + "=" + stubAnnotationValue(value), ","
+        ) + ")"
+  else result = a.getType().getName()
 }
 
-private string stubAnnotiationSimpleValue(Expr value) {
-  result = value.(FieldAccess).getField().getQualifiedName() or
-  result = value.(Literal).toString()
+private string stubAnnotationSimpleValue(Expr value) {
+  result = value.(FieldAccess).getField().getQualifiedName()
+  or
+  result = value.(Literal).getLiteral()
+  or
+  not value instanceof Literal and
```

Comment:
Probably also have to make sure that it is not a `FieldAccess` (or rather `FieldRead`) either since `CompileTimeConstantExpr` includes that as well.

Reply:
On second thought, I think we don't need to handle that much level of detail: these are stubs, so literal values aren't relevant. I changed it to just use the default value we normally use for fields and return values.

Reply:
Ok, but you probably have to special-case `String` elements because `stubDefaultValue` would use `null` for them, which is not allowed as annotation element value.